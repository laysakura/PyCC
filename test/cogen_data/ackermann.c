int ack(int m, int n)
{
  if(m == 0)
    return n + 1;
  if(n == 0)
    return ack(m - 1, 1);
  return ack(m - 1, ack(m, n - 1));
}

int c0func(int a, int b, int c, int d, int e, int f)
{
  int i;
  
  i = 0;
  while(i < b)
    {
      print_int(ack(a, i));
      print_line();
      i = i + 1;
    }
  return 0;
}
