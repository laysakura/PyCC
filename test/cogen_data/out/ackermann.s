	.text
.global	ack
	.type	ack,	@function
ack:
	pushl	%ebp
	movl	%esp, %ebp
	subl	$128, %esp
	movl	8(%ebp), %eax
	cmpl	$0, %eax
	jne	.L0
	movl	12(%ebp), %eax
	movl	%eax, -4(%ebp)
	addl	$1, -4(%ebp)
	movl	-4(%ebp), %eax
	leave
	ret
.L0:
	movl	12(%ebp), %eax
	cmpl	$0, %eax
	jne	.L1
	movl	8(%ebp), %eax
	movl	%eax, -8(%ebp)
	subl	$1, -8(%ebp)
	movl	-8(%ebp), %eax
	movl	%eax, 0(%esp)
	movl	$1, 4(%esp)
	call	ack
	movl	%eax, -12(%ebp)
	movl	-12(%ebp), %eax
	leave
	ret
.L1:
	movl	8(%ebp), %eax
	movl	%eax, -16(%ebp)
	subl	$1, -16(%ebp)
	movl	12(%ebp), %eax
	movl	%eax, -20(%ebp)
	subl	$1, -20(%ebp)
	movl	8(%ebp), %eax
	movl	%eax, 0(%esp)
	movl	-20(%ebp), %eax
	movl	%eax, 4(%esp)
	call	ack
	movl	%eax, -24(%ebp)
	movl	-16(%ebp), %eax
	movl	%eax, 0(%esp)
	movl	-24(%ebp), %eax
	movl	%eax, 4(%esp)
	call	ack
	movl	%eax, -28(%ebp)
	movl	-28(%ebp), %eax
	leave
	ret
	.size	ack, .-ack
.global	c0func
	.type	c0func,	@function
c0func:
	pushl	%ebp
	movl	%esp, %ebp
	subl	$120, %esp
	movl	$0, -4(%ebp)
.L2:
	movl	-4(%ebp), %eax
	cmpl	12(%ebp), %eax
	jge	.L3
	movl	8(%ebp), %eax
	movl	%eax, 0(%esp)
	movl	-4(%ebp), %eax
	movl	%eax, 4(%esp)
	call	ack
	movl	%eax, -8(%ebp)
	movl	-8(%ebp), %eax
	movl	%eax, 0(%esp)
	call	print_int
	movl	%eax, -12(%ebp)
	call	print_line
	movl	%eax, -16(%ebp)
	movl	-4(%ebp), %eax
	movl	%eax, -20(%ebp)
	addl	$1, -20(%ebp)
	movl	-20(%ebp), %eax
	movl	%eax, -4(%ebp)
	jmp	.L2
.L3:
	movl	$0, %eax
	leave
	ret
	.size	c0func, .-c0func
	.ident	"PyCC by @laysakura"
	.section	.note.GNU-stack,"",@progbits
