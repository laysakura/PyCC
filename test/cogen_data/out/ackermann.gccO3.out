5
13
29
61
125
253
509
1021
2045
4093
8189
16381
32765
